nuget/NativeNuget.nuspec

```diff
@@ -31,15 +31,53 @@
 	<!-- assemble files in the /nuget directory and run nuget pack from -->
 	<files>
 		<file src="LICENSE.txt" target="LICENSE.txt" />
-		<!-- Windows, Linux, macOS via netstandard -->
+		<file src="ORT_icon_for_light_bg.png" target="ORT_icon_for_light_bg.png" />
+		<file src="..\ThirdPartyNotices.txt" target="ThirdPartyNotices.txt" />
+
+		<!--
+			NOTE: If you wish to create a local nuget package for testing, update the relevant src paths below 
+			to point to the local build you wish to test, and comment out the entries for any platforms you don't 
+			need in the nuget package.
+			Run `nuget pack NativeNuget.nuspec` from this directory to produce the package for local testing.
+		-->
+		<!-- Windows and Linux via netstandard -->
 		<file src="..\out\Windows\bin\RelWithDebInfo\ortextensions.dll" target="runtimes\win-x64\native" />
+		<!-- update source paths to match the CI as the builds for the various platforms are added
+		<file src="???-win-x86\lib\ortextensions.dll" target="runtimes/win-x86/native"/>
+		<file src="???-win-arm\lib\ortextensions.dll" target="runtimes/win-arm/native"/>
+		<file src="???-win-arm64\lib\ortextensions.dll" target="runtimes/win-arm64/native"/>
+		
+		<file src="???-linux-aarch64\lib\libortextensions.so" target="runtimes/linux-arm64/native"/>		
+		<file src="???-linux-x64\lib\libortextensions.so" target="runtimes/linux-x64/native"/>
```

Comment:
@wejoncy can we fill in the various src values with the paths from the CI and check that in so the script updates the checked in file instead of generating it all dynamically?

Reply:
Yes. Would [this](https://github.com/microsoft/onnxruntime-extensions/pull/408/files#diff-42656c0cafc8b6f749f5d61e541b4d422803d710ac6024293bce190b33ea93b7R91) be good?

Reply:
Do we need to always add files dynamically vs hardcoding all the ones the CI will include?

Trying to understand what the content of `generate_file_list` equates to in terms of actual entries in the nuspec is virtually impossible. If we check in a file with all the `file` entries it's really easy for a new developer to understand what is going into the nuget package and how that is defined, so I'd really like to do that if possible. 

AFAIK there are 2 scenarios:

On the CI we assemble artifacts and extract them, with the CI controlling the directory names etc. that artifacts are extracted to. if we can use relative paths and the `file` entries are included in the checked in NativeNuget.nuspec a user could technically re-create that directory structure locally if they really wanted as they could see exactly which files are required. 

The other scenario is a user wanting to test some changes they're making locally. Typically they would just be testing one platform, but it's possible they would want to do multiple (e.g. Android and iOS). In this case the src paths are totally dependent on how the user did those individual builds. e.g. did they use build.bat or build_lib.bat for example. Due to that we really need them to provide paths for each individual library they want to add. For that to work we also need something to figure out all the things they're not providing to disable related pieces. e.g. no android AAR means the targets for android need to be excluded. Question is whether it's worth complicating this to allow a developer to update the `files` entries via a script. As it's relatively simple to edit the nuspec if all the possible entries are there to start with I'm not sure it is.

And if a user wanted to test more than a couple of platforms it would be easier for them to run the nuget packaging pipeline on their branch to get a nuget package to test with from that. 

Definitely worth the script being able to update versions and commit ids etc. though.

Reply:
Let's do only update work in the python script.